core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -167,31 +172,31 @@ def get_minimum_model_set(self, model: str | AbstractModel, include_self: bool =
             minimum_model_set = [m for m in minimum_model_set if m != model]
         return minimum_model_set
 
-    def get_model_info(self, model: str | AbstractModel) -> dict[str, Any]:
+    def get_model_info(self, model: str | ModelTypeT) -> dict[str, Any]:
         if isinstance(model, str):
             if model in self.models.keys():
                 model = self.models[model]
-        if isinstance(model, str):
             model_type = self.get_model_attribute(model=model, attribute="type")
             model_path = self.get_model_attribute(model=model, attribute="path")
             model_info = model_type.load_info(path=os.path.join(self.path, model_path))
         else:
             model_info = model.get_info()
         return model_info
+    
+    def get_model_names(self, **kwargs) -> list[str]:
+        """Get all model names that are registered in the model graph"""
+        return list(self.model_graph.nodes)
```

Comment:
Would recommend adding a docstring detail on if there are any guarantees on the ordering of the names in the output. For example, is it the order it was added to the graph, or is it alphanumeric?

Reply:
I added "in no particular order." Underlying implementation appears to be a dict, for model_graph which doesn't guarantee ordering. Moreover, even if it were, I don't think the interface should constrain this.